1Password/docusaurus-extensions
# [DEV] Docusaurus deps to 3.2.1

## Summary

Upgrade Docusaurus  deps to 3+

## Related Issues and MRs

Closing #54,
Relies on #52 

## What changed

- Upgrade `@docusaurus/core` and adjacent packages:`@docusaurus/types`, `@docusaurus/module-type-aliases`, `@docusaurus/theme-classic`, `@docusaurus/preset-classic`, `@docusaurus/utils-validation`, and `@docusaurus/plugin-content-docs`, and `@docusaurus/eslint-plugin` to ^3.2.1
- Resolve breaking changes from Docusaurus 3+ (imports + api changes)
- Resolve chalk version so that package could properly compile ([Compilation error encountered with chalk](https://github.com/nuxt/nuxt/issues/13297#issuecomment-1397294220))

## Steps for reviewer

1. Pull changes and install
```
git checkout brian/docusaurus-3.2.1
yarn install
```
2. Verify `@docusaurus/*` deps have been updated throughout the project
3. Verify our plugins build properly
```
yarn dev docusaurus-plugin-stored-data
```
```
yarn dev jest-docusaurus
```

## Additional resources

- https://docusaurus.io/docs/migration/v3
- [Compilation error encountered with chalk](https://github.com/nuxt/nuxt/issues/13297#issuecomment-1397294220)
